### DISCUSSION TITLE
Improvements to 2FA Backoffice User Experience

### DISCUSSION THREAD

**Discussion by MMasey** [umbraco/Umbraco-CMS#14585] [Features and ideas]
I've recently been playing around with implementing 2FA into a v10 site. The setup itself is really well documented, and it all works very well without too much effort.

Whilst from a functional perspective it works great I feel like there are some improvements that could be made. I tried looking at adding these as a custom package, but had trouble due to some of the 2FA code already being built into Umbraco's code base, which got me thinking maybe it makes sense to just improve it for anyone else using it.

So with that said, I have the following suggestions to improve how 2FA is presented in the backoffice

1. Add a "2FA" enabled icon to each of the user cards in the user section. This could also be done for members
2. Add a "2FA" enabled icon to the individual user details section
3. In the user details section, add which 2FA providers are available, and which ones are enabled to the user info panel
![image](https://github.com/umbraco/Umbraco-CMS/assets/6573613/eb742cf8-f9a5-47ef-9710-1d80838d354c)
4. Make it possible to enforce 2FA for users and members

It would be great to hear other peoples thoughts. I feel that with how easy it is to configure now as a baseline, I would imagine more and more companies wanted to enable this in future, so it would be great to add some improvements by default.

**Comment by lassefredslund** [comment]
I fully agree, and think it would be a lovely contribution.
Let me know if you would like a short alignment session before you start implementing.

**Reply by shearer3000** [reply]
Hello - regarding #4 "Make it possible to enforce 2FA for users": is it definitely not possible to force this already (and why not?)
